src/boot/efi/devicetree.c

```diff
@@ -133,7 +129,7 @@ EFI_STATUS devicetree_install_from_memory(
 void devicetree_cleanup(struct devicetree_state *state) {
         EFI_STATUS err;
 
-        if (!state->pages)
+        if (!state->pages || !state->orig)
```

Comment:
Shouldn't the configuration table be cleared anyway in this case? As that's the 'original' state that is being restored

Reply:
@kukrimate any suggestion on the appropriate thing to do here? Not many DTB experts around here

Reply:
This line comes from my original PoC and it is not related to being devicetree-correct. I had already more time than I should have reviewing a code base I'd never seen before (systemd-boot) and writing a PoC. I suspect I was no longer in the mood to go an double check that that InstallConfigurationTable() could tolerate a NULL pointer (given I was asking someone else to test it, I probably figured that loading the wrong config table would be easier to debug remotely than a crash in systemd-boot)!

However I just checked both EDK2 implementation and the UEFI spec. Passing a NULL to InstallConfigurationTable is just fine and does the right thing. Therefore removing this line is "the right thing to do".